### PR TITLE
SPRT-119 - Add Staking API Guide for ETH Unstaking

### PR DESCRIPTION
**Pull request by pyr0gan**
- Add a Staking API Guide for ETH Withdrawals using pre-signed exit transactions

### CODE REVIEW

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        "validator_index": "398094",
+        "signature": "0xa5ccd66f59898d3b655affe90ca328849c5993a54e4f63ed5447ce0f3bb6b5918bf72f06cbaec29af169abc433209eb30aa95d7ac1513c79887e2b552590119c32f350924e8f1dda2f2c1a5a68d801f3c150e8234493e0658031abdddcc9b2e3"
+    }
+}'
+```
+
+### Successful Response
+
+- The flow state becomes `broadcasting` when the unstaking message is awaiting confirmation on-chain, providing an `estimated_state_change_at` timestamp.
+- When the unstaking message has been confirmed, the flow state becomes `exiting` while the validator is in the withdrawal queue, then `withdrawing` while the withdrawal is being processed, and finally `unstaked`.
```
Comment (flanagansteve) [inline docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:62]: Could be worth noting, in some way, that this progression from `broadcasting` -> `unstaked` happens thanks to an async tracking job on our side

File: docs/guides/staking-api/Ethereum/unstaking/index.mdx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+sidebar_position: 3
+pagination_prev: null
+---
+
+import Link from "@docusaurus/Link";
+import img1 from "@site/static/img/staking/workflows/v1/ethereum/Workflows::V1::Ethereum::UnstakingFlow_workflow.png";
+
+## Introduction
+
+Make a withdrawal from the Ethereum staking smart contract using this flow in the Staking API!
```
Comment (flanagansteve) [inline docs/guides/staking-api/Ethereum/unstaking/index.mdx:13]: The withdrawal does not come from a smart contract FWIW. Maybe something like:

> Withdraw your staked ETH from Ethereum's consensus layer using the Staking API.

though that's too wordy

File: docs/guides/staking-api/Ethereum/unstaking/index.mdx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- `signature` &mdash; A BLS signature of the [`VoluntaryExit`](https://ethereum.github.io/beacon-APIs/#/Beacon/submitPoolVoluntaryExit) message by the private key associated with the `validator_pubkey` matching the `validator_index`.
+
+Follow the steps in this guide for the details on this process.
+
+## Flow Diagram
+
+Below is a complete diagram for the Ethereum unstaking flow:
+
+<img
+  src={img1}
```
Comment (flanagansteve) [inline docs/guides/staking-api/Ethereum/unstaking/index.mdx:28]: did we upload this img? would love to see it

**Review by flanagansteve — APPROVED** [review]

File: docs/guides/staking-api/Ethereum/unstaking/index.mdx (hunk 2 above)
Comment (yash-sharma1) [reply docs/guides/staking-api/Ethereum/unstaking/index.mdx:13]: @pyr0gan I agree with @flanagansteve.

File: docs/guides/staking-api/Ethereum/unstaking/index.mdx (hunk 3 above)
Comment (yash-sharma1) [reply docs/guides/staking-api/Ethereum/unstaking/index.mdx:28]: @flanagansteve yes we did, but you can't see it on GitHub, adding a snapshot below 
image

**Review by yash-sharma1 — APPROVED** [review]
This is good to go, just a small optional change in the wording on the `unstaking/index.mdx` page mentioned by Steve.

File: docs/guides/staking-api/Ethereum/unstaking/index.mdx (hunk 3 above)
Comment (flanagansteve) [reply docs/guides/staking-api/Ethereum/unstaking/index.mdx:28]: nice!

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    "estimated_principal_return_at": null
+  },
+  "network_code": "ethereum",
+  "chain_code": "goerli",
+  "created_at": "2023-02-23T05:16:02.187Z",
+  "updated_at": "2023-02-24T11:28:44.565Z"
+}
+```
+
+- `estimated_principal_return_at` &mdash; A timestamp tracking when the unstaked ETH will exit the validator and go to the withdrawal address.
```
Comment (coltonc3) [inline docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:91]: Exiting a validator and withdrawing the ETH are two separate processes (thanks Ethereum). 

Can simplify this with just: A timestamp tracking when the unstaked ETH will be available at the withdrawal address.

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+  "network_code": "ethereum",
+  "chain_code": "goerli",
+  "created_at": "2023-02-23T05:16:02.187Z",
+  "updated_at": "2023-02-24T13:01:55.668Z"
+}
+```
+
+### Failed Response
+
+- If at any point, the broadcast, exit transaction, or withdrawal transaction become expired &mdash; the flow state will become `failed`. Refer to the [Unstaking Flow Diagram](/guides/staking-api/ethereum/unstaking#flow-diagram) for a visual representation.
```
Comment (coltonc3) [inline docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:149]: I think we can remove "or withdrawal transaction" since technically it's not really a thing, unless @flanagansteve there is something on the slate side that could fail here?

File: docs/guides/staking-api/Ethereum/unstaking/index.mdx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+## Introduction
+
+Make a withdrawal from the Ethereum staking smart contract using this flow in the Staking API!
+
+To unstake ETH using a pre-signed exit transaction and the Staking API, you will need:
+
+- `validator_index` &mdash; The validator index that your validator was assigned when it joined the beacon chain.
+- `epoch` &mdash; A recent epoch during which the validator will exit the active set.
+- `signature` &mdash; A BLS signature of the [`VoluntaryExit`](https://ethereum.github.io/beacon-APIs/#/Beacon/submitPoolVoluntaryExit) message by the private key associated with the `validator_pubkey` matching the `validator_index`.
```
Comment (coltonc3) [inline docs/guides/staking-api/Ethereum/unstaking/index.mdx:19]: same comments as above 👍

**Review by coltonc3 — APPROVED** [review]
just a few minor changes in wording but overall looks great!

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+For more information about the upcoming Shanghai/Capella upgrade and the different types of ETH withdrawals, refer to Figment&apos;s [Ethereum Withdrawals: A Comprehensive FAQ](https://www.figment.io/resources/ethereum-withdrawals-a-comprehensive-faq-guide) and the Ethereum Foundation&apos;s [ETH Withdrawals FAQ](https://notes.ethereum.org/@launchpad/withdrawals-faq).
+
+:::
+
+### Parameters
+
+- `name` &mdash; `string` Name of the action to execute (required).
+- `inputs` &mdash; `object` (required).
+  - `epoch` &mdash; `number` &mdash; Ethereum epoch when the validator will exit (required).
```
[the remark this answers is not in the document]
Comment (flanagansteve) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:18]: To clarify for everyone - unintuitively, this value has to be for a past epoch

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 1 above)
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:62]:
```
### Successful Response

- The flow state becomes `broadcasting` when the unstaking message is awaiting confirmation on-chain, providing an `estimated_state_change_at` timestamp.
- When the unstaking message has been confirmed, the flow state becomes `exiting` while the validator is in the withdrawal queue, then `withdrawing` while the withdrawal is being processed, and finally `unstaked`.

:::info note

The progression from `broadcasting` to `unstaked` is tracked asynchronously by the Staking API, without any user interaction required.

:::
```

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 7 above)
[the remark this answers is not in the document]
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:18]:
```
  - `epoch` &mdash; `number` &mdash; Earliest epoch when a voluntary exit can be processed for the given validator. This value must be for a past epoch, not a future epoch (required).
```

?

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+  - `validator_index` &mdash; `number` &mdash; Index of the validator from which ETH will be unstaked (required).
+  - `signature` &mdash; `hexadecimal` &mdash; Ethereum withdrawal signature (required). See below for details.
+
+:::caution important
+
+The `validator_index` number is assigned to validators when they join the beacon chain. A validator index and its public key are two ways of referring to the same validator.
+Validator information is publicly available on beacon chain explorers, such as [https://beaconscan.com/validators](https://beaconscan.com/validators) or [https://beaconcha.in/validators](https://beaconcha.in/validators).
+
+The `signature` parameter is a BLS signature of the `VoluntaryExit` message by the private key associated with the `validator_pubkey` matching the `validator_index`.
```
[the remark this answers is not in the document]
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:27]: @coltonc3 What specifically are you referring to? Which above suggestion?

Is it not necessary to explain that the signature being supplied here is _not the signature included in the voluntary exit_, but rather a signature of the voluntary exit itself?

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 4 above)
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:91]: Got it 😄

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 5 above)
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:149]: @coltonc3 @flanagansteve 

```
- If at any point, the broadcast or exit transaction become expired &mdash; the flow state will become `failed`. Refer to the [Unstaking Flow Diagram](/guides/staking-api/ethereum/unstaking#flow-diagram) for a visual representation.
```
?

File: docs/guides/staking-api/Ethereum/unstaking/index.mdx (hunk 6 above)
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/index.mdx:19]: Got it 😄

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+For more information about the upcoming Shanghai/Capella upgrade and the different types of ETH withdrawals, refer to Figment&apos;s [Ethereum Withdrawals: A Comprehensive FAQ](https://www.figment.io/resources/ethereum-withdrawals-a-comprehensive-faq-guide) and the Ethereum Foundation&apos;s [ETH Withdrawals FAQ](https://notes.ethereum.org/@launchpad/withdrawals-faq).
+
+:::
+
+### Parameters
+
+- `name` &mdash; `string` Name of the action to execute (required).
+- `inputs` &mdash; `object` (required).
+  - `epoch` &mdash; `number` &mdash; Ethereum epoch when the validator will exit (required).
+  - `validator_index` &mdash; `number` &mdash; Index of the validator from which ETH will be unstaked (required).
```
[the remark this answers is not in the document]
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:19]: Got it 😄

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+  - `signature` &mdash; `hexadecimal` &mdash; Ethereum withdrawal signature (required). See below for details.
```
[the remark this answers is not in the document]
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:20]: Got it 😄

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 8 above)
[the remark this answers is not in the document]
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:27]: Got it 👍

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx [hunk 11]
```diff
@@ -15,16 +15,16 @@ For more information about the upcoming Shanghai/Capella upgrade and the differe
 
 - `name` &mdash; `string` Name of the action to execute (required).
 - `inputs` &mdash; `object` (required).
-  - `epoch` &mdash; `number` &mdash; Ethereum epoch when the validator will exit (required).
-  - `validator_index` &mdash; `number` &mdash; Index of the validator from which ETH will be unstaked (required).
-  - `signature` &mdash; `hexadecimal` &mdash; Ethereum withdrawal signature (required). See below for details.
+  - `epoch` &mdash; `number` &mdash; Earliest epoch when a voluntary exit can be processed for the given validator. This value must be for a past epoch, not a future epoch (required).
```
Comment (coltonc3) [inline docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:18]: Fairly certain it can actually be a past or current epoch. sorry took me a while to find source but here: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#voluntary-exits

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 1 above)
Comment (coltonc3) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:62]: @flanagansteve think there's a 4 epoch (~25 minutes) wait from request to exit until exit begins, which I think is worth adding in the docs. Can you confirm this is the way Staking API is handling this?

Comment (coltonc3) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:62]: getting source confirmation from Thalman.

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 11 above)
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx]: See 39f45de 😄

File: docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx (hunk 1 above)
Comment (pyr0gan) [reply docs/guides/staking-api/Ethereum/unstaking/2_submit-unstaking-data.mdx:62]: After source confirmation, and related to DEVAPIS-273: See 39f45de 😄
